easybuild/framework/easyconfig/format/one.py

```diff
@@ -312,64 +343,136 @@ def extract_comments(self, rawtxt):
         Inline comments on items of iterable values are also extracted.
         """
         self.comments = {
-            'above': {},  # comments for a particular parameter definition
+            'above': {},  # comments above a parameter definition
             'header': [],  # header comment lines
             'inline': {},  # inline comments
-            'iter': {},  # (inline) comments on elements of iterable values
-            'tail': [],
+            'iterabove': {},  # comment above elements of iterable values
+            'iterinline': {},  # inline comments on elements of iterable values
+            'tail': [],  # comment at the end of the easyconfig file
          }
 
-        rawlines = rawtxt.split('\n')
+        parsed_ec = self.get_config_dict()
 
-        # extract header first
-        while rawlines and rawlines[0].startswith('#'):
-            self.comments['header'].append(rawlines.pop(0))
+        comment_regex = re.compile(r'^\s*#')
+        param_def_regex = re.compile(r'^([a-z_0-9]+)\s*=')
+        whitespace_regex = re.compile(r'^\s*$')
 
-        parsed_ec = self.get_config_dict()
+        def clean_part(part):
+            """Helper function to strip off trailing whitespace + trailing quotes."""
+            return part.rstrip().rstrip("'").rstrip('"')
+
+        def split_on_comment_hash(line, param_key):
+            """Helper function to split line on first (actual) comment character '#'."""
+
+            # string representation of easyconfig parameter value,
+            # used to check if supposed comment isn't actual part of the parameter value
+            # (and thus not actually a comment at all)
+            param_strval = str(parsed_ec.get(param_key))
+
+            parts = line.split('#')
+
+            # first part (before first #) is definitely not part of comment
+            before_comment = parts.pop(0)
 
+            # strip out parts that look like a comment but are actually part of a parameter value
+            while parts and ('#' + clean_part(parts[0])) in param_strval:
+                before_comment += '#' + parts.pop(0)
+
+            comment = '#'.join(parts)
+
+            return before_comment, comment.strip()
+
+        def grab_more_comment_lines(lines, param_key):
+            """Grab more comment lines."""
+
+            comment_lines = []
+
+            while lines and (comment_regex.match(lines[0]) or whitespace_regex.match(lines[0])):
+                line = lines.pop(0)
+                _, actual_comment = split_on_comment_hash(line, param_key)
+                # prefix comment with '#' unless line was empty
+                if line:
+                    actual_comment = '# ' + actual_comment
```

Comment:
If lines[0] is whitespace only, then wouldn't that end up as
# <line-with-only-witespace>

Reply:
fixed by using `line.strip()`